File: packages/dvai-bridge-capacitor-llama/android/src/main/java/co/deepvoiceai/dvaibridge/llama/LlamaHandlers.kt
package co.deepvoiceai.dvaibridge.llama

import kotlinx.serialization.json.*

/**
 * Stub implementation of the OpenAI-compatible handler set for the llama backend.
 * Real inference paths land in Task 36; for now, /v1/models returns a canned
 * response and the other endpoints return 501.
 */
class LlamaHandlers(
    @Suppress("unused") private val bridge: LlamaCppBridge,
    @Suppress("unused") private val modelId: String,
) : DvaiHandlers {

    override suspend fun handleChatCompletion(body: JsonObject, ctx: HandlerContext): HandlerResponse =
        HandlerResponse.Json(501, buildJsonObject { put("error", "Not implemented yet -- Task 36") })

    override suspend fun handleCompletion(body: JsonObject, ctx: HandlerContext): HandlerResponse =
        HandlerResponse.Json(501, buildJsonObject { put("error", "Not implemented yet -- Task 36") })

    override suspend fun handleEmbeddings(body: JsonObject, ctx: HandlerContext): HandlerResponse =
        HandlerResponse.Json(501, buildJsonObject { put("error", "Not implemented yet -- Task 36") })

    override suspend fun handleModels(ctx: HandlerContext): HandlerResponse =
        HandlerResponse.Json(
            200,
            buildJsonObject {
                put("object", "list")
                putJsonArray("data") {
                    addJsonObject {
                        put("id", ctx.modelId)
                        put("object", "model")
                        put("owned_by", "dvai-bridge")
                    }
                }
            },
        )
}

File: packages/dvai-bridge-capacitor-llama/ios/Sources/DVAICapacitorLlama/Internal/LlamaHandlers.swift
// Internal/LlamaHandlers.swift
import Foundation
import DVAICapacitorLlamaObjC

/// Stub implementation. Task 36 replaces all four handler methods with real
/// llama.cpp calls. For Task 28's lifecycle test, `handleModels` returns the
/// canned model list and the other three return 501.
public final class LlamaHandlers: DVAIHandlers, @unchecked Sendable {
    private let bridge: LlamaCppBridge
    private let modelId: String

    public init(bridge: LlamaCppBridge, modelId: String) {
        self.bridge = bridge
        self.modelId = modelId
    }

    public func handleChatCompletion(body: [String: Any], ctx: HandlerContext) async throws -> HandlerResponse {
        return .json(501, ["error": "Not implemented yet — Task 36"])
    }

    public func handleCompletion(body: [String: Any], ctx: HandlerContext) async throws -> HandlerResponse {
        return .json(501, ["error": "Not implemented yet — Task 36"])
    }

    public func handleEmbeddings(body: [String: Any], ctx: HandlerContext) async throws -> HandlerResponse {
        return .json(501, ["error": "Not implemented yet — Task 36"])
    }

    public func handleModels(ctx: HandlerContext) async throws -> HandlerResponse {
        return .json(200, [
            "object": "list",
            "data": [["id": ctx.modelId, "object": "model", "owned_by": "dvai-bridge"]],
        ])
    }
}
